src/yoroi-wallets/cardano/transactionManager.test.ts

```diff
@@ -0,0 +1,148 @@
+import {fromPairs} from 'lodash'
+import DeviceInfo from 'react-native-device-info'
+
+import {storage as rootStorage, YoroiStorage} from '../storage'
+import {Transaction} from '../types'
+import {mockApi, mockedAddressesByChunks, mockedBackendConfig, mockedHistoryResponse, mockTx} from './mocks'
+import {perAddressTxsSelector, toCachedTx} from './shelley'
+import {makeMemosManager, makeTransactionManager} from './transactionManager'
+
+jest.mock('./api', () => mockApi)
+
+describe('transaction manager', () => {
+  DeviceInfo.getVersion = () => '9.9.9'
+
+  let txManager
+  beforeEach(async () => {
+    txManager = await makeTransactionManager(mockStorage, mockedBackendConfig)
+  })
+
+  afterEach(rootStorage.clear)
+
+  it('stores memos', async () => {
+    expect(txManager.getTransactions()).toEqual({[mockTx.id]: mockTx})
+
+    await txManager.saveMemo(mockTx.id, 'memo 1')
+
+    expect(txManager.getTransactions()).toEqual({[mockTx.id]: {...mockTx, memo: 'memo 1'}})
+  })
+
+  it('gets updated tx cache values', async () => {
```

Comment:
Testing that the getters are tracking the changes in tx cache state.

Through doing `perAddressTxs: transactionCache.confirmationCounts` the changes aren't tracked as can be verified by making the appropriate modifications in the tests and in tx manager

Reply:
the tests should resemble how you intend the class to be used
if you dont intend for users of this class to use selectors, dont use them in the tests

Reply:
it is no longer necessary. using snapshots.